metrics/src/main/java/io/opencensus/metrics/MetricRegistry.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.common.ToDoubleFunction;
+import io.opencensus.common.ToLongFunction;
+import io.opencensus.internal.Utils;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+
+/**
+ * Creates and manages your application's set of metrics. Exporters use the meter registry to
+ * iterate over the set of metrics instrumenting your application, and then further export each
+ * metric to the backend of choice.
+ *
+ * @since 0.16
+ */
+@ExperimentalApi
+public abstract class MetricRegistry extends MetricProducer {
+  /**
+   * Build a new long gauge to be added to the registry.
+   *
+   * <p>Must be called only once.
+   *
+   * @param name the name of the metric.
+   * @param description the description of the metric.
+   * @param unit the unit of the metric.
+   * @param obj the function argument.
+   * @param function the function to be called.
+   * @since 0.16 @ExperimentalApi
+   */
+  public abstract <T> void addLongGauge(
```

Comment:
Not sure why we need this method, given that we already have the factory method on creating `MetricDescriptor` and `Metric`?

Reply:
Chatted offline, `MetricRegistry` is mainly to be used by application users to record Gauge  metrics conveniently.